reactor-aeron-core/src/main/java/reactor/aeron/MessagePublication.java

```diff
@@ -364,7 +367,10 @@ boolean isTimeoutElapsed(Duration timeout) {
     private void resetBuffer() {
       Object oldBuffer = buffer;
       buffer = null;
-      bufferHandler.dispose(oldBuffer);
+      if (buffer != oldBuffer) {
```

Comment:
what is it? it's not clear just check it more clearly:
```
if (oldBuffer != null) {
...
```

Reply:
> what is it?

it's a java code